### ISSUE TITLE
ClassCastException during URI serialization

### ISSUE THREAD

**Issue by ghost** [cognitect-labs/aws-api#145]
### Description
This exception was observed when using `[com.cognitect.aws/kafka "801.2.686.0"]` to make an `{:op :DescribeConfigurationRevision}` request:
```clojure
   {:request
    (...
     {:op :DescribeConfigurationRevision,
      :request
      {:Arn
       "arn:aws:kafka:...",
       :Revision 1}}),
    :response
    {:cognitect.anomalies/category :cognitect.anomalies/fault,
     :cognitect.aws.client/throwable #error {
    :cause "class java.lang.Long cannot be cast to class java.lang.String (java.lang.Long and java.lang.String are in module java.base of loader 'bootstrap')"
```

At the line in question:

[ecad8a3010bc6c4dd3ac3005b15c8a6020fbee2a](https://github.com/cognitect-labs/aws-api/blob/ecad8a3010bc6c4dd3ac3005b15c8a6020fbee2a/src/cognitect/aws/protocols/rest.clj#L38)

When the `input-shape` supplied to `cognitect.aws.protocols.rest/serialize-uri` has a member that has a `:shape` of type `"__long"`, and a `"uri"` `:location`, the said exception is thrown:
```clojure
user> (cognitect.aws.protocols.rest/serialize-uri "/v1/configurations/{arn}/revisions/{revision}" {:type "structure", :members {:Arn {:shape "__string", :location "uri", :locationName "arn"}, :Revision
{:shape "__long", :location "uri", :locationName "revision"}}, :required ["Revision" "Arn"]} {:arn "arn:aws:kafka:test:123456:configuration/test/123456", :revision 1})
Execution error (ClassCastException) at cognitect.aws.util/url-encode (util.clj:190).
class java.lang.Long cannot be cast to class java.lang.String (java.lang.Long and java.lang.String are in module java.base of loader 'bootstrap')
```

### Solution
For each URI path value, covert the value to a string before encoding:
```diff
diff --git a/src/cognitect/aws/protocols/rest.clj b/src/cognitect/aws/protocols/rest.clj
--- a/src/cognitect/aws/protocols/rest.clj
+++ b/src/cognitect/aws/protocols/rest.clj
@@ -35,6 +35,7 @@
                                remove-leading-slash)
                        (some-> args
                                (get (keyword param))
+                               str
                                util/url-encode
                                remove-leading-slash))
```

**Comment by dchelimsky** [comment]
8a0a3d3165d96030bbc0f5ae6b30a25b37bbf6c5

This will be part of the next release, this week or next. Thanks!

**Comment by dchelimsky** [comment]
Released today in version 0.8.620. Thanks!
